GitUI/CommandsDialogs/SettingsDialog/Pages/GitConfigSettingsPage.cs

```diff
@@ -133,118 +144,124 @@ protected override void PageToSettings()
             }
         }
 
-        private void GlobalMergeTool_TextChanged(object sender, EventArgs e)
+        private string BrowseDiffMergeTool(string toolName, string path)
         {
-            if (IsLoadingSettings)
+            DiffMergeToolConfiguration diffMergeToolConfig = default;
+            if (_NO_TRANSLATE_GlobalDiffTool.SelectedIndex > -1)
             {
-                return;
+                diffMergeToolConfig = _diffMergeToolConfigurationManager.LoadDiffMergeToolConfig(toolName, null);
             }
 
-            var mergeTool = _NO_TRANSLATE_GlobalMergeTool.Text.Trim();
-            MergetoolPath.Text = CurrentSettings.GetValue($"mergetool.{mergeTool}.path");
-            MergeToolCmd.Text = CurrentSettings.GetValue($"mergetool.{mergeTool}.cmd");
-
-            MergeToolCmd.Enabled = !string.IsNullOrEmpty(MergeToolCmd.Text);
+            string initialDirectory = GetInitialDirectory(path, diffMergeToolConfig.Path);
 
-            MergeToolCmdSuggest_Click(null, null);
-        }
+            var filter = !string.IsNullOrWhiteSpace(diffMergeToolConfig.ExeFileName)
+                ? $"{toolName}|{diffMergeToolConfig.ExeFileName}"
+                : "*.exe;*.cmd;*.bat|*.exe;*.cmd;*.bat";
 
-        private void MergeToolCmdSuggest_Click(object sender, EventArgs e)
-        {
-            if (!EnvUtils.RunningOnWindows())
+            using (var dialog = new OpenFileDialog
             {
-                return;
-            }
-
-            var mergeToolPath = MergetoolPath.Text.Trim().Trim('"', '\'');
-
-            CurrentSettings.SetPathValue(string.Format("mergetool.{0}.path", _NO_TRANSLATE_GlobalMergeTool.Text.Trim()), mergeToolPath ?? "");
-            string exeName;
-            string exeFile;
-            if (!string.IsNullOrEmpty(mergeToolPath))
+                Filter = filter,
+                InitialDirectory = initialDirectory,
+                Title = _selectFile.Text
+            })
             {
-                exeFile = mergeToolPath;
-                exeName = Path.GetFileName(exeFile);
+                return dialog.ShowDialog() == DialogResult.OK ? dialog.FileName : path;
             }
-            else
+        }
+
+        private string GetInitialDirectory(string path, string toolPreferredPath)
+        {
+            if (!string.IsNullOrWhiteSpace(path))
             {
-                exeFile = MergeToolsHelper.FindMergeToolFullPath(ConfigFileSettingsSet, _NO_TRANSLATE_GlobalMergeTool.Text, out exeName);
+                try
+                {
+                    string initialDirectory = Path.GetDirectoryName(path);
```

Comment:
If path is a directory without a slash, the directory will be cleaned. Suggest the following:

                    if (Directory.Exists(path))
                    {
                        return PathUtil.ToNativePath(PathUtil.EnsureTrailingPathSeparator(path));
                    }